indico/modules/events/registration/client/js/form/fields/TimetableSessions.jsx

```diff
@@ -0,0 +1,296 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import timeTableURL from 'indico-url:event_registration.api_event_timetable';
+
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import {useSelector} from 'react-redux';
+import {Checkbox, Form, Accordion, Icon} from 'semantic-ui-react';
+
+import '../../../../../../rb/client/js/modules/roomList/filters/EquipmentForm.module.scss';
+
+import {
+  FinalCheckbox,
+  FinalRadio,
+  FinalField,
+  FinalInput,
+  validators as v,
+} from 'indico/react/forms';
+import {useIndicoAxios} from 'indico/react/hooks';
+import {Translate, PluralTranslate} from 'indico/react/i18n';
+import {camelizeKeys} from 'indico/utils/case';
+import {$T} from 'indico/utils/i18n';
+
+import '../../../styles/regform.module.scss';
+import {getStaticData} from '../selectors';
+
+function TimetableSessionsComponent({
+  value,
+  name,
+  sessionData,
+  onChange,
+  onFocus,
+  onBlur,
+  allowFullDays,
+  display,
+  minimum,
+  maximum,
+  ...props
+}) {
+  const markTouched = () => {
+    onFocus();
+    onBlur();
+  };
+
+  function handleValueChange(e, data) {
+    markTouched();
+    const id = data.value;
+    const newValue = value.includes(id) ? value.filter(el => el !== id) : [...value, id];
+    onChange(newValue);
+  }
+
+  function formatDate(date) {
+    const options = {
+      year: 'numeric',
+      month: 'short',
+      day: 'numeric',
+      weekday: 'short',
+    };
+    const dt = new Date(date);
+    return dt.toLocaleDateString(undefined, options);
+  }
+
+  const currentSelectionMsg = $T.gettext('You selected {0} session blocks.').format(value.length);
```

Comment:
I think it helps the user to keep track of the number of selections they have done, specially when they have collapsed some days in big events with several blocks.

I would keep it, but I agree with re-styling it, but not sure how, since I want it to be visible, but not bigger than field's title, however, I feel field's title is a bit small already, so I didn't really knew how to approach this.

Reply:
Maybe we could display the number of selected blocks for each day instead? I'm thinking sth like this:
![image](https://github.com/indico/indico/assets/8739637/1495a974-55b9-450b-9b78-e66dc3b9bde2)